### DISCUSSION TITLE
Any way to display errors full screen instead of inside iframe?

### DISCUSSION THREAD

**Discussion by antja0** [livewire/livewire#5529] [1. Help] [answered]
Errors are created inside modals. For example upon exception the 500 page will show inside iframe, that appears as a popup modal on top of the page. I would like to redirect the user to 500 page instead.

Is there any way to disable this behaviour?

<img width="955" alt="image" src="https://user-images.githubusercontent.com/47383357/214707488-a2c0d283-e6d7-4c3e-8f5a-7dc00a2295e1.png">

**Comment by joshhanley** [comment, marked as answer]
@antja0 not currently. Any particular reason you want to redirect them to the 500 page instead of displaying the modal? The idea with the modal is they get the error but then they can click away and try whatever they were doing again.

**Reply by antja0** [reply]
Hi @joshhanley,

The idea of modal is good and I can see it being handy in some situations. Although for "small errors" I tend to use [dispatchBrowserEvent](https://laravel-livewire.com/docs/2.x/events#browser) instead to create a small notification for the user.

Just to inform you, here are some use cases / reasons for full page errors that I've come across:

1. I'm authenticating with [Laravel policies](https://laravel.com/docs/10.x/authorization#generating-policies) and [AuthorizesRequests](https://laravel-livewire.com/docs/2.x/authorization) trait for full page livewire components, and I want the user to see full page 403 error (and 403 status code), especially when the user is coming to the website from external link.

2. In policy I have `return Response::denyWithStatus(402);` if users subscription has expired. I would like them to be able to directly manage their payment settings by clicking a link. This won't really work because links will open up inside the modal iframe.

3. I want to create a special 500 error page with support ticket form, and after submitting a ticket I want to redirect the user to the ticket. But it will open inside the modal iframe. + if another error is thrown inside the modal eg. when creating the ticket, it will create another error modal inside the modal.

4. On general I want to put links to the error page, for example link to the home page. However if user clicks this link, it will open the link inside the modal.

5. Overall Google recommends refraining from creating iframes. I personally don't see any problem with them, but I tend to avoid them unless I need them. There are all kinds of small issues, like for example I cannot use `X-Frame-Options: DENY` now.

6. It's a bit confusing (at least for me it was) because native Laravel works by opening a separate page when doing eg. `abort(404)`

**Reply by joshhanley** [reply]
@antja0 yeah that makes sense. At this stage, we're not going to change it as we're working on V3. But once V3 is released, if you feel it could still be handled better, please raise this again or better yet submit a PR with a proposed solution. 🙂

**Reply by antja0** [reply]
OK!
Hey, that V3 sounds exiting! Can't wait, I'm loving livewire so far!

**Reply by timgavin** [reply]
I second this. I created a custom `/errors/500.blade.php` and when it shows in a modal it's breaking the layout because it's such a narrow viewport.

**Comment by masihgh** [comment]
guys please fix this :))

**Comment by masihgh** [comment]
## you can do this trick:

```css
#livewire-error{
     padding: 0px !important;
}
   ```
  
  > It's work but still its an iframe :(

**Comment by DaW-S** [comment]
This one works fine for me in Livewire V3. 

In layout
```
<script>
    document.addEventListener('livewire:init', () => {
        Livewire.hook('request', ({ fail }) => {
            fail(({ status, preventDefault }) => {
                if (status != 419) {
                    window.location.href = `/error/${status}`
                    preventDefault()
                }
            })
        })
    })
</script>
```

In routes
```
Route::get('/error/{code}', function ($code) {
    $view = "errors.$code";

    if (View::exists($view)) {
        return response()->view($view, [], (int) $code);
    }

    return response()->view('errors.500', [], 500);
})->whereNumber('code');
```
